docs/extending_wsimod.md

```diff
@@ -0,0 +1,95 @@
+# Extending and customising WSIMOD functionality
+
+WSIMOD has many features built-in, but undoubtedly, to model real life scenarios, you will need to customise it to your specific needs.
+
+WSIMOD offers 4 ways of doing this customisation of increased complexity - and flexibility: custom orchestration, overrides, patches and custom classes.
+
+## Custom orchestration
+
+WSIMOD allows to define the order in which the sequence of actions for each node must take place. This orchestration can be customised within the `orchestration` section of the config file. For example, the following orchestration will run the simulation with two actions: first it will run the `infiltrate` method of the `Groundwater` node, and then the `make_discharge` method of the `Sewer` node.
+
+```yaml
+orchestration:
+- Groundwater: infiltrate
+- Sewer: make_discharge
+```
+
+If no `orchestration` is provided, then a default sequence is used. Check the [Orchestration section](./orchestration.md) for more details on how to customise this.
+
+## Overrides
+
+Overrides are the next way of customising the behaviour of nodes or arcs. They enable specifying the value of one or more parameters of a specific - existing - node or arc. These overrides are specified in the config file for the model under a `overrides` section, and therefore they need to be objects that can be parsed in yaml - strings, floats, etc.
+
```

Comment:
```suggestion

Because many parameters require other states to change when they are changed, our recommendation is that any parameter changes are either changed directly in the `config` file or inside the `overrides` section of the config file (which offers a convenient location to include all changes made to a base `config`). Because of this, if you are changing parameters on-the-fly, we **strongly recommend** using the [`apply_overrides`](reference-model.md#wsimod.orchestration.model.Model.apply_overrides) functionality of the [`Model`](reference-model.md#wsimod.orchestration.model.Model) object.

```

Reply:
@liuly12 Is this OK??

Reply:
OK with me!